empress/clumpr/Statistics.py

```diff
@@ -0,0 +1,98 @@
+# Statistics.py
+# Computes statistics and creates cost histogram
+# Ran Libeskind-Hadas, June 2020
+
+# Performs Monte Carlo simulation to compute empirical p-value and histogram
+# of solution costs.
+# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC379178/
+
+from clumpr import DTLReconGraph
+import random
+import matplotlib
+matplotlib.use('tkagg') # need this so plt.show() works
+import matplotlib.pyplot as plt
+
+from typing import List, Dict, Tuple
+
+def trials(host_tree: dict, parasite_tree: dict, phi: dict, dup_cost: float, transfer_cost: float, loss_cost: float, num_trials: int) -> list:
+    """
+    :param host_tree: dictionary representation of host tree
+    :param parasite_tree: dictionary of parasite tree
+    :param phi: dictionary representation of parasite tip to host tip mapping
+    :param dup_cost: float duplication cost
+    :param transfer_cost: float transfer cost
+    :param loss_cost: float loss cost
+    :param num_trials: int number of trials in Monte Carlo simulation
+    :return: list of floating point costs of reconciliations of the Monte Carlo samples
+    """
+   
+    costs = list()  # List of costs of random trials
+    parasites = list(phi.keys())
+    hosts = list(phi.values())
+    for t in range(num_trials):
+        random_phi = create_random_phi(phi)
+        for p in parasites:
+            h = random.choice(hosts)
+            random_phi[p] = h
+        _, cost, _, _ = DTLReconGraph.DP(host_tree, parasite_tree, random_phi, dup_cost, transfer_cost, loss_cost)
+        costs.append(cost)
+    return costs
+
+def create_random_phi(phi : dict) -> dict:
+    """
+    :param phi: dictionary representation of parasite tip to host tip mapping
+    :return: a dictionary of parasite tips to host tips that preserves the degree of
+        of the host tips in phi
+    """
+    random_phi = {}
+    parasites = list(phi.keys())
+    hosts = phi.values()
+    unique_hosts = list(set(hosts))
+    parasite_count = {}   # keys are hosts, values are the number of parasites on that host wrt phi
+    # computer parasite_count values
+    for p in parasites:
+        h = phi[p]
+        if h in parasite_count: parasite_count[h] += 1
+        else: parasite_count[h] = 1
+    for h in unique_hosts:
+        parasite_samples = random.sample(parasites, parasite_count[h])
+        for p in parasite_samples: 
+            random_phi[p] = h
+            parasites.remove(p)
+    return random_phi
+
+def plot_histogram(MPR_cost: int, costs: list):
```

Comment:
I think the variable name should be snake_case. Could you change this to `mpr_cost`?

Reply:
Fixed.